### DISCUSSION TITLE
Getting only specific columns while importing file

### DISCUSSION THREAD

**Discussion by fzdunek** [spatie/simple-excel#168] [Q&A]
Hello!
I have a question about importing a file - I have a file with unnecessary columns that unnecessarily burden memory. Is it possible to 
SimpleExcelReader pass only specific columns that it will define beforehand?

**Comment by variablesign** [comment]
> Hello! I have a question about importing a file - I have a file with unnecessary columns that unnecessarily burden memory. Is it possible to SimpleExcelReader pass only specific columns that it will define beforehand?

> Behind the scenes generators are used, so memory usage will be low, even for large files.

In one of my projects, I tried to import 200k rows with Laravel Excel package and the queue process took 2 hours to complete with an average memory usage of 1GB. I tried to import the same file with this package, and it took 30 minutes with an average memory usage of 80MB so am sure there wouldn't be much burden on your memory.

_The test was run on a 7th Gen Intel i5 laptop with 128GB SSD and 8GB RAM._
